src/Task/Recurrence.ts

```diff
@@ -76,12 +84,12 @@ export class Recurrence {
                 // Assuming due date has the highest priority.
                 let referenceDate: Moment | null = null;
                 // Clone the moment objects.
-                if (dueDate) {
-                    referenceDate = window.moment(dueDate);
-                } else if (scheduledDate) {
-                    referenceDate = window.moment(scheduledDate);
-                } else if (startDate) {
-                    referenceDate = window.moment(startDate);
+                if (occurrence.dueDate) {
+                    referenceDate = window.moment(occurrence.dueDate);
+                } else if (occurrence.scheduledDate) {
+                    referenceDate = window.moment(occurrence.scheduledDate);
+                } else if (occurrence.startDate) {
+                    referenceDate = window.moment(occurrence.startDate);
```

Comment:
Those repeated `occurrence.thing...` references make me wonder whether that code might now be better placed in the `Occurrence` class?